pandas/_libs/lib.pyx

```diff
@@ -2161,11 +2195,23 @@ def maybe_convert_numeric(
     if seen.check_uint64_conflict():
         return values
 
+    # This occurs since we disabled float nulls showing as null in anticipation
+    # of seeing ints that were never seen. So then, we return float
+    if allow_null_in_int and seen.null_ and not seen.int_:
+        seen.float_ = True
+
     if seen.complex_:
         return complexes
     elif seen.float_:
+        if seen.null_ and convert_to_masked_nullable:
+            return (floats, mask.view(np.bool_))
         return floats
     elif seen.int_:
+        if seen.null_ and convert_to_masked_nullable:
+            if seen.uint_:
+                return (uints, mask.view(np.bool_))
```

Comment:
hmm can we simply make the return value *always* a tuple, e.g. (values, mask) where mask could be None

Reply:
This adds unnecessary complexity to maybe_convert_numeric usages that don't use nullable support. @jbrockmendel WDYT?

Reply:
i agree with jreback that returning a tuple in a subset of cases is undesirable.

My first choice would be to handle this outside of lib.pyx entirely.  Second choice would be to import the appropriate masked array here (despite the fact that in general we try to avoid that pattern).  Third choice would be to change the return type